### PR TITLE
add hybrid fsdp ShardingStrategy

### PR DESCRIPTION
**Pull request by pkumc**
# What does this PR do?

Add [hybrid fsdp ShardingStrategy](https://pytorch.org/docs/stable/fsdp.html#torch.distributed.fsdp.ShardingStrategy) to Trainer.

## Who can review?
- trainer: @muellerzr and @pacman100

### CODE REVIEW

**Comment by pkumc** [comment]
@pacman100 Do you have time to review this PR?

**Comment by pkumc** [comment]
@muellerz Do you have time to review this PR?

**Comment by pkumc** [comment]
> @pacman100 Do you have time to review this PR?

@muellerzr Could you review this PR?

**Review by pacman100 — COMMENTED** [review]
Hello, FSDP new features are now supported via Accelerate's integration as the options are getting more and more and we don't want to bloat Trainer arguments: See this https://huggingface.co/docs/transformers/main_classes/trainer#using-accelerate-launcher-with-trainer.

File: src/transformers/trainer.py [hunk 1]
```diff
@@ -432,6 +432,10 @@ def __init__(
                 self.fsdp = ShardingStrategy.SHARD_GRAD_OP
             elif FSDPOption.NO_SHARD in args.fsdp:
                 self.fsdp = ShardingStrategy.NO_SHARD
+            elif FSDPOption.HYBRID_SHARD in args.fsdp:
```
Comment (pacman100) [inline src/transformers/trainer.py:435]: These are no longer used with PyTorch. They are only used with FSDP XLA integration. For PyTorch, it uses Accelerate's integration via the codeblock here: 

https://github.com/huggingface/transformers/blob/38e2633f80a4924bf613b0240622492beee4cfcc/src/transformers/training_args.py#L1643-L1673

Comment (pkumc) [reply src/transformers/trainer.py:435]: ok. I see, thx.

**Pull request closed without merging**